moto/route53/responses.py

```diff
@@ -81,6 +81,9 @@ def list_or_create_hostzone_response(self, request: Any, full_url: str, headers:
                 delegation_set_id=delegation_set_id,
             )
             template = Template(CREATE_HOSTED_ZONE_RESPONSE)
+            headers = {
+                "Location": f"https://route53.amazonaws.com/2013-04-01/hostedzone/{new_zone.id}"
```

Comment:
shouldn't this use `full_url + '/' + new_zone.id` instead, to respect the potential custom domain ? I'm wondering whether the current patch will be working for localstack...

Reply:
Hi @stof! If I understand the architecture of LocalStack correctly, the `full_url` will always become `route53.amazonaws.com` for their invocations. So that wouldn't make a difference.

It is a bit tricky to return URL's, because when Moto is run inside Docker, whatever endpoint (and port) we expose inside the container may be very different than what is actually accessible by the user. Maybe LS can rewrite the Location to a URL that makes sense for them, such as `localhost.localstack.cloud`?